operator/operator_test.go

```diff
@@ -178,7 +182,36 @@ func (e *environment) WaitPhase() {
 	for {
 		select {
 		case head := <-heads:
-			if e.enclaveParameters.IsLastPhaseBlock(head.Number.Uint64() - 1) {
+			if e.enclaveParameters.IsLastPhaseBlock(head.Number.Uint64()) {
+				log.Warn("Phase ended in block #", head.Number.Uint64())
+				time.Sleep(200 * time.Millisecond)
+				return
+			}
+		case <-ctx.Done():
+			e.T.Fatal("context: ", ctx.Err())
+		case err := <-sub.Err():
+			e.T.Fatal("header subscription: ", err)
+		}
+	}
+}
+
+func (e *environment) WaitBlocks(n uint) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(blockTime*n+1)*time.Second)
+	defer cancel()
+	heads := make(chan *types.Header)
+	sub, err := e.operator.ethClient.SubscribeNewHead(ctx, heads)
+	if err != nil {
+		e.T.Fatal("subscribing to header: ", err)
+	}
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case head := <-heads:
+			n--
+			if n == 0 {
+				log.Warn("Phase ended in block #", head.Number.Uint64())
```

Comment:
I wouldn't put a `Warn` log here. Also, `WaitBlock` doesn't know why it is called, so it cannot say "Phase ended".